client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java

```diff
@@ -205,6 +206,39 @@ public void testGetWithBody() throws IOException {
         bodyTest("GET");
     }
 
+    /**
+     * Ensure that basePath works as expected.
+     */
+    public void testBasePath() throws IOException {
+        final String basePath = "base/" + randomAsciiOfLengthBetween(1, 5) + "/";
+        final int statusCode = randomStatusCode(getRandom());
+
+        HttpContext context = null;
+
+        try (final RestClient client =
+                RestClient.builder(new HttpHost(httpServer.getAddress().getHostString(), httpServer.getAddress().getPort()))
+                    .setBasePath(basePath).build()) {
+
+            context = httpServer.createContext("/" + basePath + statusCode, new ResponseHandler(statusCode));
+
+            for (String method : getHttpMethods()) {
+                Response esResponse;
+                try {
+                    esResponse = client.performRequest(method, "/" + statusCode);
```

Comment:
Good point.

> Maybe this test should be better isolated and have only its own context defined?

I'll add a unique suffix so that there's no way the context exists without being given the status code _and_ something else.